### ISSUE TITLE
Add expvar handler to zpages extension

### ISSUE THREAD

**Issue by yurishkuro** [open-telemetry/opentelemetry-collector#11081]
**Is your feature request related to a problem? Please describe.**

Go stdlib's expvar provides some useful information about Go runtime, and sometimes it is used by various libraries that could be imported by OTEL components to report their own internal state. But the HTTP handler is not exposed by default so there is no way to introspect that state of the libraries in the collector.

**Describe the solution you'd like**

Since zpage extension is the agreed mechanism for introspection of the running collector, and already provides a webserver, we can register expvar handler on that server (also mentioned in https://github.com/open-telemetry/opentelemetry-collector/issues/2155#issuecomment-728167103).

The zpage extension config can have a flag that enables expvar, which could be off by default.

**Describe alternatives you've considered**
There could be a separate expvar extension just to create a new webserver and open a port, but it seems like an overkill and not aligned with zpages as the main introspection facility.

**Comment by HongChenTW** [comment]
Hi @yurishkuro, I'd like to try this one, it seems a good opportunity to understand the zpage extension implementation.

**Comment by yurishkuro** [comment]
@HongChenTW go for it. I temporarily created an internal extension in Jaeger as a workaround https://github.com/jaegertracing/jaeger/pull/5986

**Comment by HongChenTW** [comment]
Hi @yurishkuro , after digging into code, I found that I could register the expvar handler to expose the expvar pages at [here](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.109.0/extension/zpagesextension/zpagesextension.go), with some adjustments(declare the `expvarz` path variable, add port, on/off configs, etc.), but I'm not sure if I'm in the right path.

e.g.  `zPagesMux.Handle(path.Join("/debug", "expvarz"), expvar.Handler())`

But I also noticed other components in zpages extention draw html pages like [tracez](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/zpages/internal/templates/traces.html) or [featurez](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.109.0/service/internal/zpages/templates/features_table.html), should I also design a html template to render the expvar, could you give some advises?

Thanks!

**Comment by yurishkuro** [comment]
@HongChenTW I would start with just the default, out of the box `expvar.Handler()`. It returns JSON which is already pretty readable, and there are probably tools today that can scrape that format if someone needs it. I personally just used a browser plugin for JSON rendering so that I can expand/collapse some of the more complex structures supported by the expvar package.

**Comment by arsenalzp** [comment]
Hello colleagues,
Are this issue suitable for newcomers who wish to familiarize oneself with this great OpenTelemetry project?

**Comment by yurishkuro** [comment]
Yes

**Comment by dridi-walid** [comment]
Hi, I'd like to work on it.

**Comment by HongChenTW** [comment]
Hi @yurishkuro , could you help to check this PR https://github.com/open-telemetry/opentelemetry-collector/pull/11217 if is mergeable?

I'm not sure whether the implementation is suitable for the requirements in this issue, also, since this PR was created few months ago, could you help to guide how to safely update the PR branch with the latest commit in main branch, I want to prevent to reference on the others' commits and show up the actions I made in their issues & PRs.

e.g.
![image](https://github.com/user-attachments/assets/7eac5647-710f-4c4b-a60a-c460952adc52)

Thanks!

**Comment by yurishkuro** [comment]
yes, it looks like a good PR. It's easier to use the Merge button on GitHub to avoid pulling other commits. If you want to re-create, you could start a fresh branch from the current `main` and use `git cherry-pick` to select the commits you need from the old branch.

**Comment by HongChenTW** [comment]
Thanks for the suggestion, I'll create another branch.
